app/parsers/bulkrax/csv_parser.rb

```diff
@@ -10,19 +10,38 @@ def self.parser_fields
       }
     end
 
-    def run
-      create_works
-    end
-
     def records(_opts = {})
-      CSV.foreach(
+      @records ||= CSV.foreach(
```

Comment:
does this work or does the @records become a foreach that would need to be rewound to actually be useful?

Reply:
I think this is still an open question?

Reply:
It does work. Tested with CSV Parser creating one collection and three works.